### PR TITLE
[git] Allow fetch without update repo

### PR DESCRIPTION
**Pull request by valeriocos**
This PR allows to fetch the commits of a repo without performing any update against the target repo. This new feature is enable by setting the fetch param `no_update` to true (by default is disabled). Tests have been created accordingly.

This change is needed to allow ELK to detect the commits that have been deleted in the repo and remove them from the target index (see https://github.com/chaoss/grimoirelab-elk/pull/538)

### CODE REVIEW

File: perceval/backends/core/git.py [hunk 1]
```diff
@@ -348,6 +355,9 @@ def setup_cmd_parser():
         group.add_argument('--latest-items', dest='latest_items',
                            action='store_true',
                            help="Fetch latest commits added to the repository")
+        group.add_argument('--no-update', dest='no_update',
```
Comment (sduenas) [inline perceval/backends/core/git.py]: Probably, `--latest-items` and `--no-update` should be mutually exclusive. To use both at the same time doesn't make sense.

File: tests/test_git.py [hunk 2]
```diff
@@ -584,6 +584,86 @@ def test_fetch_latest_items_from_empty_repository(self):
 
         shutil.rmtree(new_path)
 
+    def test_fetch_no_update(self):
+        """Test whether all commits are fetched from a Git repository"""
```
[the remark this answers is not in the document]
Comment (valeriocos) [reply tests/test_git.py]: why the comment is wrong?

**Comment by valeriocos** [comment]
@sduenas should I address your comments, rework the PR to implement and use `git rev-list` (see last commit) or you prefer to have two PRs one for the param `update` and the other one with the command `git rev-list`? Note that also in the related PR in ELK I added a [commit](https://github.com/chaoss/grimoirelab-elk/pull/538) to discuss which solution was better

File: tests/test_git.py [hunk 3]
```diff
@@ -758,6 +835,15 @@ def test_setup_cmd_parser(self):
         self.assertEqual(parsed_args.git_path, '/tmp/gitpath')
         self.assertEqual(parsed_args.uri, 'http://example.com/')
         self.assertEqual(parsed_args.branches, ['master', 'testing'])
+        self.assertFalse(parsed_args.no_update)
```
Comment (sduenas) [inline tests/test_git.py]: I would put the code below in a new tests. It's hard to understand why this is failing. It's failing because `--no-update` and `--latest-items` are mutually exclusive. Having this in a test named like `test_mutual_exclusive_update` or something like that will make live easier. Someone would read the test and will know what's going on.

File: tests/test_git.py [hunk 4]
```diff
@@ -584,6 +584,81 @@ def test_fetch_latest_items_from_empty_repository(self):
 
         shutil.rmtree(new_path)
 
+    def test_fetch_no_update(self):
+        """Test whether the fetch method properly works when the no update flag is set"""
```
Comment (sduenas) [inline tests/test_git.py]: I would reword it like "Test whether new commits aren't fetched when no_update flag is set"
